whelktool/scripts/cleanups/2019/10/lxl-2737-remove-redundant-blank-languages/script.groovy

```diff
@@ -0,0 +1,78 @@
+import datatool.util.LanguageLinker
+import datatool.util.LanguageMapper
+import datatool.util.Statistics
+
+/*
+ * Replace blank language nodes with links
+ *
+ * See LXL-2737 for more info.
+ *
+ */
+
+OBSOLETE_CODES = ['9ss', '9sl']
+
+PrintWriter scheduledForUpdate = getReportWriter("scheduled-for-update")
+
+// These are cases that won't be handled by metadata (definitions) improvements that we still want to clean up
+substitutions = [
+        'jap'                             : 'jpn',
+        'mongoliskt språk'                : 'mongoliska språk',
+        'mongoliska'                      : 'mongoliska språk',
+        'mongoliska språket'              : 'mongoliska språk',
+        'latviešu val'                    : 'lettiska',
+        'latviešu valodā'                 : 'lettiska',
+        'suomi'                           : 'finska',
+        'česky'                           : 'tjeckiska',
+        'ruotsi'                          : 'svenska',
+
+        'grekiska (nygrekiska)'           : 'nygrekiska',
+        'franska (fornfranska)'           : 'fornfranska',
+        'tyska (medelhögtyska)'           : 'medelhögtyska',
+        'engelska (medelengelska)'        : 'medelengelska',
+        'engelska (fornengelska)'         : 'fornengelska',
+        'tyska (lågtyska)'                : 'lågtyska',
+        'nederländska (medelnederländska)': 'medelnederländska',
+        'norska (nynorsk)'                : 'nynorsk',
+        'franska (medelfranska)'          : 'medelfranska',
+        'tyska (medellågtyska)'           : 'medellågtyska',
+        'arabiska (judearabiska)'         : 'judearabiska',
+        'samiska (nordsamiska)'           : 'nordsamiska',
+        'samiska (lulesamiska)'           : 'lulesamiska'
+]
+
+linker = new LanguageLinker(buildLanguageMap())
+
+selectByCollection('bib') { bib ->
+    try {
+        def (record, thing, work) = bib.graph
+        if (!((String) work['@id']).endsWith('#work')) {
```

Comment:
If we need to semantically check if this is a work, either see if its type is a subclass of Work, or if thing links to it with instanceOf.

(Also, which case cropped up to require this check? It may be an indicator of something being wrong...)

Reply:
It was an quick and dirty way to guard against (theoretical) instances without local works. Checking type would be more correct.